### ISSUE TITLE
check-requires fails

### ISSUE THREAD

**Issue by samth** [racket/macro-debugger#29]
All uses of `raco check-requires` now error like this:

```
[samth@huor:~/.../collects/racket/private (master) plt] raco check-requires stx.rkt
(file "stx.rkt"):
ERROR in (file "stx.rkt")
application: procedure does not expect an argument with given keyword
  procedure: for-subnodes
  given keyword: #:recur/phase-up
  arguments...:
   (ecte #743=#<syntax:stx.rkt:4:0 (module stx (quote #%kernel) (#%declare #:cross-phase-persistent) (define-values (identifier?) (lambda (p) (if (syntax? p) (symbol? (syntax-e p)) #f))) (define-values (stx-null?) (lambda (p) (if (null? p) #t (if (syntax? ...
   #:recur #<procedure:recur>
   #:recur/phase-up #<procedure:recur/phase-up>
  context...:
   /home/samth/sw/plt/racket/collects/racket/private/kw.rkt:1532:14
   /home/samth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/analysis/private/get-references.rkt:19:0: deriv->refs
   /home/samth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/analysis/check-requires.rkt:120:0: check-requires
   /home/samth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/analysis/check-requires.rkt:288:3: for-loop
   ...cket/cmdline.rkt:191:51
   (submod "/home/samth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/analysis/check-requires.rkt" main): [running body]
```

**Comment by samth** [comment]
#29 gets it to run on `racket/private/stx`. But it then fails like this:

```
[samth@huor:~/.../collects/racket/private (master) plt] raco check-requires cond.rkt
(file "cond.rkt"):
ERROR in (file "cond.rkt")
match: no matching clause for #<procedure:node>
  location...:
   ...amth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/model/deriv-util.rkt:106:2
  context...:
   /home/samth/sw/plt/racket/collects/racket/match/runtime.rkt:24:0: match:error
   /home/samth/sw/plt/extra-pkgs/macro-debugger/macro-debugger-text-lib/macro-debugger/analysis/private/get-references.rkt:24:2: recur
   [repeats 1 more time]
```

**Comment by samth** [comment]
Now fixed by 9eac95bbd29d4535b9ea21196624f7ca3cfb3be0 among others.
